src/Command/Auth/OAuth2Command.php

```diff
@@ -0,0 +1,202 @@
+<?php
+namespace Platformsh\Cli\Command\Auth;
+
+use CommerceGuys\Guzzle\Oauth2\AccessToken;
+use GuzzleHttp\Client;
+use Platformsh\Cli\Command\CommandBase;
+use Platformsh\Cli\Service\Url;
+use Platformsh\Cli\Util\PortUtil;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+
+class OAuth2Command extends CommandBase
+{
+    protected $hiddenInList = true;
+
+    protected function configure()
+    {
+        $service = $this->config()->get('service.name');
+        $this->setName('auth:oauth2')
+            ->setDescription('Log in to ' . $service . ' via OAuth2')
+            ->addOption('force', 'f', InputOption::VALUE_NONE, 'Log in again, even if already logged in');
+        Url::configureInput($this->getDefinition());
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        // Disable the API token for this command.
+        if ($this->api()->hasApiToken()) {
+            throw new \Exception('Cannot log in: an API token is set');
+        }
+        // Login can only happen during interactive use.
+        if (!$input->isInteractive()) {
+            throw new RuntimeException('Non-interactive login not supported');
+        }
+        $connector = $this->api()->getClient(false)->getConnector();
+        if (!$input->getOption('force') && $connector->isLoggedIn()) {
+            $this->stdErr->writeln('You are already logged in.');
+            return 0;
+        }
+
+        $port = PortUtil::getPort(5000, null, 5005);
```

Comment:
Not sure why `+5` only?

Reply:
Needed (for now) a list of known redirect URLs to validate on the server